### PR TITLE
Heartbeat a list of nodes every 6 hrs

### PR DESCRIPTION
**Pull request by eyakubovich**
Platform will timeout nodes w/o an update in 24 hours.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -43,7 +45,7 @@ async fn main() -> Result<()> {
     let config = Config::load(args.config, None)?;
 
     std::env::set_var("RUST_LOG", config.log_level());
-    pretty_env_logger::init();
+    pretty_env_logger::init_timed();
```
Comment (crawford) [inline src/main.rs]: This is going to cause timestamps to show up twice in most logging systems.

File: src/platform.rs [hunk 2]
```diff
@@ -64,23 +64,33 @@ impl Client {
             .upsert_clusters(pb::UpsertClustersRequest {
                 clusters: vec![cluster],
             })
-            .await?;
+            .await
+            .map_err(|e| anyhow!("{}", e.message()))?;
```
Comment (crawford) [inline src/platform.rs:68]: Why is the map_err needed? Does the error message get lost somewhere along the line?

File: src/kube_state.rs [hunk 3]
```diff
@@ -268,12 +268,19 @@ impl Resources {
         }
     }
 
-    fn upsert_node(&mut self, node: Node) -> Machine {
+    fn upsert_node(&mut self, node: Node) -> Option<Machine> {
```
Comment (crawford) [inline src/kube_state.rs]: This method name threw me off. It's no longer idempotent if it's returning an Option, plus I tend to think of a database when I hear "upsert".

**Comment by crawford** [comment]
Also, could you add the detail from the pull request comment to the commit itself. It'll be nice to be able to go back through the history to figure out why we did things the way we did.

File: src/platform.rs (hunk 2 above)
Comment (eyakubovich) [reply src/platform.rs:68]: I use `anyhow::Result` everywhere and so somewhere at the top level, I would print the error. But `tonic::Status` impl of `Display` interface just printed all the internal fields (probably via `Debug`) so it ended up too messy. I just want the message.

File: src/main.rs (hunk 1 above)
Comment (eyakubovich) [reply src/main.rs]: Good point but only once it goes through the SIEM. The problem is that with `kubectl logs` or when debugging, I don't get any timestamps.

But I'll take it out and will later add something where I can override the format via an env var.

**Review by crawford — APPROVED** [review]

**Pull request merged**